idaes/apps/grid_integration/coordinator.py

```diff
@@ -60,95 +62,22 @@ def _register_custom_commandline_options(self):
-                     dest='track_horizon',
-                     type='int',
-                     default=48)
-        prescient.plugins.add_custom_commandline_option(opt)
-
-        opt = Option('--bidding-generator',
-                     help="Specifies the generator we derive bidding strategis for.",
-                     action='store',
-                     dest='bidding_generator',
-                     type='string',
-                     default='102_STEAM_3')
-        prescient.plugins.add_custom_commandline_option(opt)
-
-        opt = Option('--bidding',
-                     help="Invoke generator strategic bidding when simulate.",
-                     action='store_true',
-                     dest='bidding',
-                     default=False)
-        prescient.plugins.add_custom_commandline_option(opt)
-
-        opt = Option('--deviation-weight',
-                     help="Set the weight for deviation term when tracking",
-                     action='store',
-                     dest='deviation_weight',
-                     type='float',
-                     default=30)
-        prescient.plugins.add_custom_commandline_option(opt)
-
-        opt = Option('--ramping-weight',
-                     help="Set the weight for ramping term when tracking",
-                     action='store',
-                     dest='ramping_weight',
-                     type='float',
-                     default=20)
-        prescient.plugins.add_custom_commandline_option(opt)
-
-        opt = Option('--cost-weight',
-                     help="Set the weight for cost term when tracking",
-                     action='store',
-                     dest='cost_weight',
-                     type='float',
-                     default=1)
-        prescient.plugins.add_custom_commandline_option(opt)
-
-        opt = Option('--rts_gmlc_data_dir',
-                     help="the relative path to rts gmlc data set",
-                     action='store',
-                     dest='rts_gmlc_data_dir',
-                     type='str',
-                     default='./RTS-GMLC/RTS_Data/SourceData/')
-        prescient.plugins.add_custom_commandline_option(opt)
-
-        opt = Option('--price_forecast_file',
-                     help="the relative path to price forecasts",
-                     action='store',
-                     dest='price_forecast_file',
-                     type='str',
-                     default='../../prescient/plugins/price_forecasts/lmp_forecasts_concat.csv')
-        prescient.plugins.add_custom_commandline_option(opt)
+        config.declare('bidding_generator',
+                       ConfigValue(domain = str,
+                                   description = 'Specifies the generator we derive bidding strategis for.',
+                                   default = None)).declare_as_argument(f'--{key}.bidding-generator')
+        ## How to access this option?
+        # options.plugin.{key}.bidding_generator
```

Comment:
`plugin_config.bidding_generator` -- I would refactor this class slightly and *store* the `plugin_config` passed into `register_plugins` somewhere. You probably also don't need the `_register_*_callbacks` methods anymore.

The `declare_as_argument` method just has to do with how the command line option is created. Using the `key` in the name is optional.

Reply:
Thanks Ben! I just pushed a small change to store the `plugin_config` as a property. I think I should get rid of the `bidding_generator` option, because the `coordinator` should get this information from both `bidder` and `tracker`.

The `_register_*_callbacks` methods make it a little easier for me to manage the plugin functions, in case we have more than one callbacks at some plugin points.